src/Appwrite/Platform/Workers/Migrations.php

```diff
@@ -233,99 +254,120 @@ protected function generateAPIKey(Document $project): Document
     }
 
     /**
-     * @param Document $project
-     * @param Document $migration
-     * @param Log $log
-     * @return void
      * @throws Authorization
      * @throws Conflict
      * @throws Restricted
      * @throws Structure
      * @throws \Utopia\Database\Exception
+     * @throws Exception
      */
-    protected function processMigration(Document $project, Document $migration, Log $log): void
+    protected function processMigration(Document $migration, Log $log): void
     {
-        /**
-         * @var Document $migrationDocument
-         * @var Transfer $transfer
-         */
-        $migrationDocument = null;
-        $transfer = null;
+        $project = $this->project;
         $projectDocument = $this->dbForConsole->getDocument('projects', $project->getId());
         $tempAPIKey = $this->generateAPIKey($projectDocument);
 
+        $transfer = $source = $destination = null;
+
         try {
-            $migrationDocument = $this->dbForProject->getDocument('migrations', $migration->getId());
-            $migrationDocument->setAttribute('stage', 'processing');
-            $migrationDocument->setAttribute('status', 'processing');
-            $log->addBreadcrumb(new Breadcrumb("debug", "migration", "Migration hit stage 'processing'", \microtime(true)));
-            $this->updateMigrationDocument($migrationDocument, $projectDocument);
+            $migration = $this->dbForProject->getDocument('migrations', $migration->getId());
 
-            $log->addTag('type', $migrationDocument->getAttribute('source'));
+            if (
+                $migration->getAttribute('source') === SourceAppwrite::getName() ||
+                $migration->getAttribute('destination') === DestinationAppwrite::getName()
+            ) {
+                $credentials = $migration->getAttribute('credentials', []);
 
-            $source = $this->processSource($migrationDocument->getAttribute('source'), $migrationDocument->getAttribute('credentials'));
+                $credentials['projectId'] = $credentials['projectId'] ?? $projectDocument->getId();
+                $credentials['endpoint'] = $credentials['endpoint'] ?? 'http://appwrite/v1';
+                $credentials['apiKey'] = $credentials['apiKey'] ?? $tempAPIKey['secret'];
 
-            $source->report();
+                $migration->setAttribute('credentials', $credentials);
+            }
 
-            $destination = new DestinationsAppwrite(
-                $projectDocument->getId(),
-                'http://appwrite/v1',
-                $tempAPIKey['secret'],
-            );
+            $migration->setAttribute('stage', 'processing');
+            $migration->setAttribute('status', 'processing');
+            $log->addBreadcrumb(new Breadcrumb('debug', 'migration', "Migration hit stage 'processing'", \microtime(true)));
```

Comment:
I think breadcrumbs were removed and not supported anymore. Please check

Reply:
Seems like no one is using it 👍 
Removed